modules/engines/engine-xsjs/src/main/resources/META-INF/dirigible/exports/XSJSLibExportsGenerator.mjs

```diff
@@ -8,32 +8,36 @@ export class XSJSLibExportsGenerator {
   processedArtefactsTable = null;
 
   constructor(stateTableParams) {
-    this.processedArtefactsTable = new XSJSLibArtefactStateTable(
+    this.processedArtefactsTable = new XSJSLibStateTable(
       stateTableParams.name,
       stateTableParams.schema
     );
   }
 
-  run(targetPath) {
-    const targetResource = repository.getResource(targetPath);
-    const targetCollection = repository.getCollection(targetPath);
-
-    if(targetResource.exists()) {
-      this._checkTableAndGenerateExportsIfNeeded(targetResource);
+  run(targetRegistryPath, targetRegistryPathType) {
+    if(targetRegistryPathType == "ExistentXSJSLibFile") {
```

Comment:
It's not ok to hardcode this here. Isn't it better to wrap the resource/collection into a Java class with properties: 
```
isCollection: boolean
container: Collection | Resource
```
And pass it from the Java side. You wouldn't need multiple `getResource/getCollection` calls

Reply:
In [5cb9542](https://github.com/SAP/xsk/pull/1651/commits/5cb9542d23925a742b305e58a5538279c30d19f0) i removed the XSJSLibSynchronizerRegistryPathType class and now there is XSJSLibSynchronizerRegistryEntity. It wraps IEntity which can be both a collection and resource and now an instance of that object is passed to the js context. This really saves a few getCollection/getResource calls on the startup of the recursion(in the publish folder case) and in single file export generation (on publish of single file)